client/request/select.go

```diff
@@ -107,6 +109,16 @@ func (s *Select) validateGroupBy() []error {
 	return result
 }
 
+func (s *Select) ToSubscriptionSelect(docID, cid string) Selection {
```

Comment:
todo: All the other `ToFoo` funcs clone the source object - this is mutating it - it could be quite an unpleasent side-effect to debug.  Please return a new, clean object instead of the old one.